collect_app/src/main/java/org/odk/collect/android/geo/GoogleMapFragment.java

```diff
@@ -462,9 +463,18 @@ private void loadReferenceOverlay() {
             referenceOverlay = this.map.addTileOverlay(new TileOverlayOptions().tileProvider(
                 new GoogleMapsMapBoxOfflineTileProvider(referenceLayerFile)
             ));
+            switch_google_map_labels("off");
+        } else {
+            switch_google_map_labels("on");
         }
     }
 
+    // Turns off google map labels for offline layers to work correctly
+    private void switch_google_map_labels(String state) {
```

Comment:
Generally we should avoid using comments. Comments should be used if it's a tricky case that requires an explanation. In 99% cases you can just come up with a better method name. In this case it would be for example `setLabelsVisibility()`; ahh and we don't use _ in method names.